list.js

```diff
@@ -1,7 +1,71 @@
 (function() {
-    var events = [];
+    var ALL_EVENTS = [];
+    var FILTERED_EVENTS = [];
+
     var ITEMS_ON_PAGE = 6;
 
+    var index = lunr(function() {
+        this.ref('id');
+        this.field('title', {boost: 10});
+        this.field('description', {boost: 5});
+    });
+
+    /**
+     * Given an array of all events and a search query,
+     * return an array of search results.
+     */
+    var filterEvents = function(allEvents, q) {
+        var results = index.search(q);
+
+        var searchResults = [];
+        for (var r in results) {
+            var e = allEvents[results[r].ref];
+            searchResults.push(e);
+        }
+
+        return searchResults;
+    };
+
+    var doSearch = function(events) {
+        var $el = $('#calendarList');
+        var q = $('#q').val();
+
+        $el.empty();
+        $el.show();
+        $el.append('<div class="arrow"></div>');
+        $el.append(
+            $('<h2>RESULTS FOR: "' + q + '"</h2>')
+        );
+
+        FILTERED_EVENTS = filterEvents(ALL_EVENTS, q);
+        if (FILTERED_EVENTS.length === 0) {
+            $el.append('<div class="q-no-item">Unfortunately, there are ' +
+                'no results matching what you\'re looking for in ' +
+                'the Columbia Film Glossary content.</div>');
+        } else {
+            refreshEvents(FILTERED_EVENTS, 1);
+        }
+        return false;
+    };
+
+    var clearSearch = function() {
+        $('#calendarList').empty();
+        $('#calendarList').hide();
+    };
+
+    $(document).ready(function() {
+        $('#search').click(doSearch);
```

Comment:
I'm moving all this stuff into the `$(document).ready()` block we already have on line 130.